### ISSUE TITLE
[ISSUE] Error creating SP on Azure when using profile instead of explicit host

### ISSUE THREAD

**Issue by alexott** [databrickslabs/terraform-provider-databricks#915]
When trying to create SP in Azure workspace and using profile instead of explicit host, the creation fails with following error:

```
╷
│ Error: display_name is required for service principals in Databricks on AWS
│
│   with databricks_service_principal.sp,
│   on vm.tf line 44, in resource "databricks_service_principal" "sp":
│   44: resource "databricks_service_principal" "sp" {
│
```

It looks like another place affected by lazy initialization of authentication part.

**Comment by nfx** [comment]
@alexott Do you think it makes sense to fix it or just document it?... the simplest fix is to skip customize diff if host is empty

**Comment by alexott** [comment]
I'll fix it - I often use profiles, so it will be less error prone for myself :-)

**Comment by nfx** [comment]
🤗

**Comment by nfx** [comment]
possibly fixed it in #965 - i tried to use cli profiles with E2 workspaces and had couple of o_O moments
